core-bundle/src/Resources/contao/library/Contao/Config.php

```diff
@@ -244,37 +247,48 @@ public function save()
 			$strFile .= "\n" . $this->strBottom . "\n";
 		}
 
-		$strTemp = md5(uniqid(mt_rand(), true));
+		$strTemp = Path::join(TL_ROOT, 'system/tmp', md5(uniqid(mt_rand(), true)));
 
 		// Write to a temp file first
-		$objFile = fopen(TL_ROOT . '/system/tmp/' . $strTemp, 'w');
+		$objFile = fopen($strTemp, 'w');
 		fwrite($objFile, $strFile);
 		fclose($objFile);
 
 		// Make sure the file has been written (see #4483)
-		if (!filesize(TL_ROOT . '/system/tmp/' . $strTemp))
+		if (!filesize($strTemp))
 		{
 			\System::log('The local configuration file could not be written. Have your reached your quota limit?', __METHOD__, TL_ERROR);
 
 			return;
 		}
 
+		$fs = new Filesystem();
+
 		// Adjust the file permissions (see #8178)
-		$this->Files->chmod('system/tmp/' . $strTemp, \Config::get('defaultFileChmod'));
+		$fs->chmod($strTemp, self::get('defaultFileChmod'));
+
+		// Create the destination path of the localconfig.php
+		$strDestination = Path::join(TL_ROOT, 'system/config/localconfig.php');
+
+		// Get the realpath in case it is a symlink
+		if ($fs->exists($strDestination))
+		{
+			$strDestination = realpath($strDestination);
```

Comment:
Using `$strDestination = realpath($strDestination) ?: $strDestination` would save use the `$fs->exists()` call.